app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/DisplayDatasetActivity.kt

```diff
@@ -131,15 +131,24 @@ class DisplayDatasetActivity : AppCompatActivity() {
 
     }
 
-    private fun checkIntentExtras(extras : Bundle?){
-        datasetId = if (extras != null && extras["dataset_id"] is String) {
-            intent.getStringExtra("dataset_id")!!
-        } else {
-            "kitchen utensils"
+    private fun checkIntentExtras(extras: Bundle?) {
+        lifecycleScope.launch {
+            datasetId = if (extras != null && extras["dataset_id"] is String) {
+                intent.getStringExtra("dataset_id")!!
+            } else {
+                "uEwDkGoGADW4hEJoJ6BA"
```

Comment:
Why is this? Should it not show nothing or a message to inform the user instead?

Reply:
This is because when we launch the activity, no intent is launched for the moment since all those activities are going to be linked with DataOverview which will give the "dataset id" intent. I decided to hardcode the DS id for the demo but of course this will be deleted when every activity will be linked. Sorry should have written a comment.

Reply:
Okay, no worries